h2/src/main/org/h2/table/TableFilter.java

```diff
@@ -265,15 +265,38 @@ public void setPlanItem(PlanItem item) {
         if (nestedJoin != null) {
             if (item.getNestedJoinPlan() != null) {
                 nestedJoin.setPlanItem(item.getNestedJoinPlan());
+            } else {
+                setScanIndexes(nestedJoin);
             }
         }
         if (join != null) {
             if (item.getJoinPlan() != null) {
                 join.setPlanItem(item.getJoinPlan());
+            } else {
+                setScanIndexes(join);
             }
         }
     }
 
+    /**
+     * Set all missing indexes to scan indexes recursively.
+     *
+     * @param table filter to start from
+     */
+    private void setScanIndexes(TableFilter filter) {
```

Comment:
seems a little odd to have this method take a param instead of being an instance method of TableFilter?